test/test.js

```diff
@@ -6889,3 +6890,53 @@ exports['use'] = {
         test.done();
     }
 };
+
+exports['streams can be used as orchestrator tasks'] = function (test) {
+    var isReady = false;
+    var s = _();
+    runTask(function () { return s; }, function () {
+        test.ok(isReady, "task finished too early");
+        test.done();
+    });
+    setTimeout(function () {
+        isReady = true;
+        s.end();
+    }, 10);
+};
+
+exports['only gutless streams and pipelines are writable'] = function (test) {
+    test.ok(_().writable, "gutless stream should be writable");
+    test.ok(_.pipeline(_.map(function (x) { return x + 1; })).writable, "pipelines should be writable");
+    test.ok(!_([]).writable, "empty stream should not be writable");
+    test.ok(!_([1, 2, 3]).writable, "non-empty stream should not be writable");
+    test.ok(!_(function (push, next) {}).writable, "generator stream should not be writable");
+    test.ok(!_().map(function (x) { return x + 1; }).writable, "mapped stream should not be writable");
+    test.done();
+};
+
+exports['ended streams should be neither readable nor writable'] = function (test) {
+    var s = _();
+    s.resume();
+    test.ok(s.readable);
+    test.ok(s.writable);
+    s.write(1);
+    test.ok(s.readable);
+    test.ok(s.writable);
+    s.end();
+    test.ok(!s.readable);
+    test.ok(!s.writable);
+    test.done();
+};
+
+exports['ended pipelines should be neither readable nor writable'] = function (test) {
```

Comment:
> ended pipelines should be neither readable nor writable'

Can you move this to the pipeline suite of tests (around line 6316)?

> 'only gutless streams and pipelines are writable'

Move to constructor suite (line 525).

> 'streams can be used as orchestrator tasks'

Add a " (issue #438)" to the end.